code/game/objects/items/weapons/harvester.dm

```diff
@@ -0,0 +1,208 @@
+//All the harvester weapons go in here
+
+#define VALI_CODEX "<b>Reagent info:</b><BR>\
+	All chems will do 60% armor-piercing damage, and also:<BR>\
+	Bicaridine - Heals brute and stamina. Channel the heal for a larger heal, works better with medical skill.<BR>\
+	Kelotane - Set your target aflame and sunder their armor<BR>\
+	Tramadol - Slow your target for 1 second<BR>\
+	Tricordrazine - Shatter your targets armor for 3 seconds<BR>\
+	<BR>\
+	<b>Tips:</b><BR>\
+	> Needs to be connected to the Vali system to collect green blood. You can connect it though the Vali system's configurations menu.<BR>\
+	> Filled by liquid reagent containers or pills. Emptied by using an empty liquid reagent container.<BR>\
+	> Press your unique action key (SPACE by default) to load a single-use of the reagent effect after the blade has been filled up."
+
+//Vali weapon base
+/obj/item/weapon/harvester
+	name = "generic vali weapon"
+	desc = "just your everyday harvester weapon."
+	icon = 'icons/obj/items/vali.dmi'
+	item_icons = list(
+		slot_l_hand_str = 'icons/mob/inhands/weapons/vali_left.dmi',
+		slot_r_hand_str = 'icons/mob/inhands/weapons/vali_right.dmi',
+	)
+	flags_atom = CONDUCT
+	flags_equip_slot = ITEM_SLOT_BELT
+	force = 10
+	throwforce = 10
+	sharp = IS_SHARP_ITEM_BIG
+	edge = 1
+	w_class = WEIGHT_CLASS_BULKY
+	attack_verb = list("attacked", "slashed", "stabbed", "sliced", "torn", "ripped", "diced", "cut")
+
+	var/codex_info = VALI_CODEX
+
+/obj/item/weapon/harvester/get_mechanics_info()
+	. = ..()
+	. += jointext(codex_info, "<br>")
+
+/obj/item/weapon/harvester/update_overlays()
+	. = ..()
+	
+
+//Vali Sword
+/obj/item/weapon/harvester/sword
+	name = "\improper HP-S Harvester blade"
+	desc = "TerraGov Marine Corps' experimental High Point-Singularity 'Harvester' blade. An advanced weapon that trades sheer force for the ability to apply a variety of debilitating effects when loaded with certain reagents. Activate after loading to prime a single use of an effect. It also harvests substances from alien lifeforms it strikes when connected to the Vali system."
+	icon_state = "vali_sword"
+	//item_state = "vali_sword"
+	force = 60
+	attack_speed = 12
+
+/obj/item/weapon/harvester/sword/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/harvester)
```

Comment:
Is there every a `/obj/item/weapon/harvester` that doesn't get the harvester component?

Reply:
The claymore has different args ``AddComponent(/datum/component/harvester, 60, TRUE)``

Reply:
Then use like `LateInit` and override it in the claymore